### ISSUE TITLE
Crash with suggest emojis by tag typing in Discord

### ISSUE THREAD

**Issue by MithicSpirit** [AnySoftKeyboard/AnySoftKeyboard#2785]
### Steps to reproduce
1. Enable "Suggest Emojis by tag typing"
2. Open Discord and go to any text channel
3. Type `:`
4. Select one of the emojis suggested by Discord (not by the keyboard); this will fill in the rest of the name of the emoji
5. Press any key on the keyboard

### Expected behavior
The keyboard should realize that stuff was completed externally and continue working normally.

### Actual behavior
The keyboard appears to crash but gets stuck on-screen, even if I hit the back or home buttons. This can be fixed by switching to a different keyboard, waiting a bit, then switching back. I presume this would also happen on other apps that somehow complete text with colons but Discord is the only one that I know that does this.

**Android OS version:** 11

**Device manufacturer and model:** Samsung Galaxy S10

**List of installed add-ons (like languages, or themes):** Brazillian Portuguese, Greek

**Comment by menny** [comment]
Oh. Interesting. Which version of AnySoftKeyboard are you running?

**Comment by MithicSpirit** [comment]
I am on v1.10-r4. (sorry for the late reply, I've been procrastinating on email recently).

**Comment by alchemiker** [comment]
Yeah, Discord and ASK are not friends. I use Discord in the browser on my phone and it produced doubled words or crashes of the keyboard, but some updates ago — whether from Discords or ASK's side — it got much better.

If you want to try whether the auto-completion-issue of emojis is Discord-specific, you can try web.whatsapp.com or even github.com here — it also suggests emojis after typing a colon, like this 🤓 but at least at my keyboard this doesn't produce a crash.

**Comment by raeokay** [comment]
Have been having the same issue on Android 10, though I can't escape the frozen keyboard without either rebooting or finding a way to force stop the app. It happens infrequently, but often enough to be a pain. Only in Discord so far and only when doing : before doing server-specific emojis. 

Am also on v1.10-r4.

**Comment by jackred** [comment]
Same behavior, also with v1.10-r4.
The only fix I found is to force stop the app. If I try to switch keyboard as proposed above, ASK get stuck in the middle of the screen and block the view. Sometimes it's able to recover but rarely.

It happens when completing any emoji (server specific, but also Discord one like :car:). Note, contrary to what is said in the issue, if space is pressed, it won't crash. It will actually cancel the auto-completion of the emoji made by discord and insert a space.
```
":emoj"
// press auto complete by Discord
":emojiDiscord:"
// press space
":emoj "
```

Also, if you move the cursor in the part that was auto completed ("iDiscord" in the example above)and try to write, it will also crash.
If you move the cursor anywhere else (before the ":" for example) and write something, then come back, it won't crash nor erase the completion. Changing keyboard, or making it disappear (slide down) won't make crash or erase the completion once you type again.  

If you attempt to past after using Discord auto completion of emoji, it won't crash nor erase the auto completed part neither. It seems to be really just typing. 
Also, if you use the auto complete feature for emoji, but you only have the last colon missing (":emojiDiscord" > ":emojiDiscord:") it won't crash when you type typing afterward.

Hope this help

**Comment by LogoiLab** [comment]
Can we get a status update on this? This is still an issue in v1.11-r1
